subsys/bluetooth/controller/hci_driver.c

```diff
@@ -32,6 +32,20 @@ static K_THREAD_STACK_DEFINE(recv_thread_stack, CONFIG_BLECTLR_RX_STACK_SIZE);
 static K_THREAD_STACK_DEFINE(signal_thread_stack,
 			     CONFIG_BLECTLR_SIGNAL_STACK_SIZE);
 
+
+/* It should not be possible to set CONFIG_BLECTRL_SLAVE_COUNT larger than
+ * CONFIG_BT_MAX_CONN. Kconfig should make sure of that, this assert is to
+ * verify that assumption. */
+BUILD_ASSERT(CONFIG_BLECTRL_SLAVE_COUNT <= CONFIG_BT_MAX_CONN);
+
+#define BLECTRL_MASTER_COUNT (CONFIG_BT_MAX_CONN - CONFIG_BLECTRL_SLAVE_COUNT)
+
+BUILD_ASSERT((IS_ENABLED(CONFIG_BT_CENTRAL) && (CONFIG_BLECTRL_SLAVE_COUNT > 0)) ||
+			 !IS_ENABLED(CONFIG_BT_CENTRAL));
```

Comment:
You really do not need !IS_ENABLED part here - it does not change anything.

Reply:
It does. If you remove it and have `CONFIG_BT_CENTRAL=n` and `CONFIG_BLECTRL_SLAVE_COUNT=0`, you will get

```c
BUILD_ASSERT( false && (0 > 0) );
```
resulting in `BUILD_ASSERT(false)` and a build assert. 

With the `!IS_ENABLED`:

```c
BUILD_ASSERT( false && (0 > 0) || !false );
```
resulting in `BUILD_ASSERT(true)` and no assert.

Reply:
@grochu, you're right :man_facepalming:

Reply:
Right, but it still can be simplified to:
```
BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_CENTRAL) || (CONFIG_BLECTRL_MASTER_COUNT > 0))
```

Reply:
Done.